service.go

```diff
@@ -0,0 +1,81 @@
+package gomicro
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/micro/go-micro/client"
+	"github.com/micro/go-micro/context"
+	"github.com/micro/go-micro/server"
+)
+
+type service struct {
+	opts Options
+}
+
+func newService(opts ...Option) Service {
+	options := newOptions(opts...)
+
+	options.Client = &clientWrap{
+		options.Client,
+		context.Metadata{
+			HeaderPrefix + "From-Service": options.Server.Config().Name(),
+		},
+	}
+
+	return &service{
+		opts: options,
+	}
+}
+
+func (s *service) Client() client.Client {
+	return s.opts.Client
+}
+
+func (s *service) Server() server.Server {
+	return s.opts.Server
+}
+
+func (s *service) String() string {
+	return "go-micro"
```

Comment:
This is basically inline with all the other packages that implement an interface. Someone could implement a version of the service interface and call it something like "dancannon". The service name used for registration is actually embedded in the Server. Need to figure out a nice simple way to initialise that too.